### DISCUSSION TITLE
How to get an understanding of the in- and output shapes?

### DISCUSSION THREAD

**Discussion by JohannesBauer97** [open-mmlab/mmdetection#9749] [Q&A]
I'm searching for the right docs/code to understand the input and outputs of the models.
For example, the RetinaNet model for object detection, after converting it to my backend CoreML using mmdeploy, I can see it has the following input and output shapes:

* **Input**: (1, 3, 800, 1344)
* **Output**: Float32 & Int32 (one dimension for `label` and `dets`)

![foto](https://user-images.githubusercontent.com/15695124/217882414-d5203035-b1e9-4dd6-ae66-c6212e217e82.png)

I guess the in- and outputs aren't changed when converting to different models, so CoreML should be just an example.

For the input shape, the 3 is probably the color channels, 800 and 1344 height and width. But how can I find out what's the first (1) dimension for? 

Also for the output shape `dets` I would have expected `x, y, w, h` and not just one dimension.

**tl;dr:** Where to find a description of the input and output shapes for the detection models?
